src/liballoc/rc.rs

```diff
@@ -354,22 +353,22 @@ impl<T> Rc<T> {
     /// ```
     #[inline]
     #[stable(feature = "rc_unique", since = "1.4.0")]
-    pub fn try_unwrap(this: Self) -> Result<T, Self> {
-        if Rc::strong_count(&this) == 1 {
+    pub fn try_unwrap(self) -> Result<T, Self> {
```

Comment:
You are changing the signature of a stable function. It probably was made an associated function instead of a method for a reason.

Reply:
Oh, wait, `this: Self` also allows method call syntax? I have never seen this...

Reply:
No, actually, I was right. `x.try_unwrap()` does not work, and that is probably deliberate. We generally avoid `self` methods on smart pointers.

Reply:
I thought that it was just the old way of writing signatures of `self` methods; it definitely seems to work the same way.

Reply:
Please drop anything related to changing functions from being associated functions to methods. This is a deliberate design decision that is even [explicitly documented](https://doc.rust-lang.org/beta/std/rc/struct.Rc.html).